samcli/commands/build/command.py

```diff
@@ -335,3 +370,47 @@ def _get_mode_value_from_envvar(name: str, choices: List[str]) -> Optional[str]:
         raise click.UsageError("Invalid value for 'mode': invalid choice: {}. (choose from {})".format(mode, choices))
 
     return mode
+
+
+def _process_env_var(container_env_vars: list) -> Dict:
+    """
+    Parameters
+    ----------
+    container_env_vars : list
+        the list of command line env vars received from --container-env-vars flag
+        Each input format needs to be either function specific format (FuncName.VarName=Value)
+        or global format (VarName=Value)
+
+    Returns
+    -------
+    dictionary
+        Processed command line environment variables
+    """
+    processed_env_vars: Dict = {}
+
+    for env_var in container_env_vars:
+        if "." in env_var:
+            test_target = env_var.split(".")
+            if len(test_target) != 2 or not test_target[0].strip() or not test_target[1].strip():
+                LOG.error("Invalid command line --container-env-vars input %s, skipped", env_var)
+                continue
+            function, variable = env_var.split(".")
+            test_target = variable.split("=")
+            if len(test_target) != 2 or not test_target[0].strip() or not test_target[1].strip():
+                LOG.error("Invalid command line --container-env-vars input %s, skipped", env_var)
+                continue
+            key, value = variable.split("=")
+            if not processed_env_vars.get(function):
+                processed_env_vars[function] = {}
+            processed_env_vars[function][key] = value
+        else:
+            test_target = env_var.split("=")
+            if len(test_target) != 2 or not test_target[0].strip() or not test_target[1].strip():
+                LOG.error("Invalid command line --container-env-vars input %s, skipped", env_var)
+                continue
+            key, value = env_var.split("=")
+            if not processed_env_vars.get("Parameters"):
+                processed_env_vars["Parameters"] = {}
+            processed_env_vars["Parameters"][key] = value
+
+    return processed_env_vars
```

Comment:
same as the `_make_env_vars()`, this can be simplified in a waterfall manner:

```py
def _process_env_var(container_env_vars: list) -> Dict:
    processed_env_vars: Dict = {}

    for env_var in container_env_vars:
        location_key = 'Parameters'
        pure_env_var = env_var
        if "." in env_var:
            test_target = env_var.split(".")
            if len(test_target) != 2 or not test_target[0].strip() or not test_target[1].strip():
                LOG.error("Invalid command line --container-env-vars input %s, skipped", env_var)
                continue
            location_key, pure_env_var = env_var.split(".")
        
        test_target = pure_env_var.split("=")
        if len(test_target) != 2 or not test_target[0].strip() or not test_target[1].strip():
            LOG.error("Invalid command line --container-env-vars input %s, skipped", env_var)
            continue
        key, value = pure_env_var.split("=")
        if not processed_env_vars.get(location_key):
            processed_env_vars[location_key] = {}
        processed_env_vars[location_key][key] = value
    return processed_env_vars
```

Reply:
thanks! refactoring this way is tested and worked as expected